src/ReactiveUI/ViewLocator.cs

```diff
@@ -93,31 +95,27 @@ public DefaultViewLocator(Func<string, string> viewModelToViewFunc = null)
         public IViewFor ResolveView<T>(T viewModel, string contract = null)
             where T : class
         {
-            var view = this.AttemptViewResolutionFor(viewModel.GetType(), contract);
+            var view = GetCandidateViewModelTypes(viewModel)
```

Comment:
I agree it reads better, but I will say that I started with an enumerable in my own implementation, but then removed it to avoid allocs. I haven't double-checked, but I'm pretty sure the previous implementation doesn't alloc (except `Type` objects, which I believe will be cached by the runtime) whereas this one will alloc with every view resolution.

Do we care? I'm not sure.